ElectronicObserver/Window/Dialog/DialogLocalAPILoader2.cs

```diff
@@ -211,15 +283,85 @@ private void ExecuteAPI(string filename)
 
 	}
 
+	private void ExecuteSortieRecordApi(string filename)
+	{
+		Debug.Assert(SortieRecord is not null);
+
+		string[] values = filename.Split(" ");
+
+		int recordId = int.Parse(values[0]);
+		ApiFileType apiFileType = Enum.Parse<ApiFileType>(values[1]);
+		string apiName = values[2];
+
+		if (!APIObserver.Instance.APIList.ContainsKey(apiName)) return;
+
+		ApiFile apiFile = ApiFilesBeforeSortie.Concat(SortieRecord.ApiFiles).First(f => f.Id == recordId);
 
+		if (apiFile.Name is "api_port/port")
+		{
+			string? savedPortResponseJson = LoadApiResponse("api_port/port");
+
+			if (savedPortResponseJson is not null)
+			{
+				ApiResponse<ApiPortPortResponse> savedPortResponse = JsonSerializer
+					.Deserialize<ApiResponse<ApiPortPortResponse>>(savedPortResponseJson[7..])
+					?? throw new NotImplementedException();
+
+				ApiResponse<ApiPortPortResponse> apiFilePortResponse = JsonSerializer
+					.Deserialize<ApiResponse<ApiPortPortResponse>>(apiFile.Content)
+					?? throw new NotImplementedException();
+
+				apiFilePortResponse.ApiData.ApiShip = savedPortResponse.ApiData.ApiShip;
+				apiFilePortResponse.ApiData.ApiLog = savedPortResponse.ApiData.ApiLog;
+				apiFilePortResponse.ApiData.ApiNdock = savedPortResponse.ApiData.ApiNdock;
+
+				apiFilePortResponse.ApiData.ApiDeckPort = SortieRecord.FleetData.Fleets
+					.Select((f, i) => f switch
+					{
+						null => null,
+
+						_ => new FleetDataDto
+						{
+							ApiId = i + 1,
+							ApiShip = f.Ships.Select(s => s.DropId ?? 0).ToList(),
+							ApiMission = [0, 0, 0, 0],
+						},
+					})
+					.OfType<FleetDataDto>()
+					.ToList();
+
+				apiFile.Content = JsonSerializer.Serialize(apiFilePortResponse, JsonSerializerOptions);
+			}
+		}
+
+		switch (apiFileType)
```

Comment:
maybe add default case, no idea how it behaves without it when running into something that isn't request or response (which shouldn't happen of course)